### DISCUSSION TITLE
Ленты

### DISCUSSION THREAD

**Discussion by picolino** [habralab/garnet-team#218] [Идеи]
Поскольку GARNet - в первую очередь социальная сеть, ей не хватает "социального" функционала.
К такому функционалу можно отнести наличие централизованных мест, вокруг которых пользователи смогут объединяться и общаться. Для нас этими местами являются, конечно, команды и проекты.

### Лента команд

Лента команды состоит из:
- markdown-постов от участников команды
- потока активностей, происходящих с командой

Посты в ленту может писать любой участник команды.
Редактировать и удалять пост может только создатель поста и владелец команды.
Посматривать ленту команд могут все.
Посты и активности показываются пагинацией (skip/take).

Поток активностей включает в себя события:
- Создание команды
- Добавление\удаление участников из команды
- Вступление и выход из участия в проектах
- ...

Пост и активность имеют дату и время создания (использовать аудитную информацию). Обязательно возвращать их в ответе

#### Техническая информация
Делаем отдельное получение постов и активностей (располагаем в отдельной коллекции в монге и делаем 2 разных запроса).

### Лента проектов

TBD

**Comment by picolino** [comment]
Если успеем - можно сделать комментарии под постами

**Comment by gfg7** [comment]
> * markdown-постов от участников команды

со стороны сервиса что требуется? я так понимаю, это фронт особые символы будет отображать по-разному
или в соответствие с спец знаком будем делить хранить текст как блоки и способом отображения этого блока

**Reply by picolino** [reply]
Мы храним чисто маркдаун - дальше фронт сам разберется как отображать

**Comment by gfg7** [comment]
> Поток активностей включает в себя события

Активности будут иметь ссылки?
К примеру, в команду добавился новый пользователь - можно перейти по ссылке на его профиль, нажав на активность или другое действие

**Reply by picolino** [reply]
Да, но их отображение - фронтовая темка
Мы только предоставляем идентификаторы и типы объектов, в которых происходит активность
